src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -79,7 +79,20 @@ public class ValidationVisitor
         /// <returns><c>true</c> if the object is valid, otherwise <c>false</c>.</returns>
         public bool Validate(ModelMetadata metadata, string key, object model)
         {
-            if (model == null && key != null)
+            return Validate(metadata, key, model, skipNullAtTopLevel: true);
+        }
+
+        /// <summary>
+        /// Validates a object.
+        /// </summary>
+        /// <param name="metadata">The <see cref="ModelMetadata"/> associated with the model.</param>
+        /// <param name="key">The model prefix key.</param>
+        /// <param name="model">The model object.</param>
+        /// <param name="skipNullAtTopLevel">If <c>true</c>, regards <c>null</c> as a valid top-level value, regardless of other validation metadata.</param>
+        /// <returns><c>true</c> if the object is valid, otherwise <c>false</c>.</returns>
+        public bool Validate(ModelMetadata metadata, string key, object model, bool skipNullAtTopLevel)
```

Comment:
In what kind of scenarios is this important?

Reply:
Other validation code paths (e.g., from `TryUpdateModelAsync`, or anyone directly calling the pre-existing `Validate` overload here) rely on this top-level-null-skipping logic, but validating at the top level is something we do want in the case of params/props.

Reply:
So is the difference here that we want to validate `[Required]` on the top level for parameter binding, and not for `TryValidateModel`?

This seems fine, just struggling to think of a more semantically meaningful name for the parameter. `skipNullAtTopLevel` describes what it does not what it means

Reply:
OK, renamed to `alwaysValidateAtTopLevel`. I think that's as semantic as I can get with this.